packages/bot-skeleton/src/scratch/utils/index.js

```diff
@@ -112,23 +122,34 @@ export const load = (block_string, drop_event, showIncompatibleStrategyDialog) =
     }
 
     try {
-        const event_group = `dbot-load${Date.now()}`;
+        const is_collection = xml.hasAttribute('collection') && xml.getAttribute('collection') === 'true';
+        const event_group = is_collection ? 'load_collection' : `dbot-load${Date.now()}`;
```

Comment:
We should add the `Date.now()` to both `load_collection` and `dbot-load` names. Imagine loading two consecutive collection strategies, a single undo would undo both imports.

Reply:
ahh, you're right!